src/DesignDocComponents.ts

```diff
@@ -2,3 +2,4 @@ import "./Globals";
 import "./Root";
 import "./SideBar";
 import "./SideNav";
+export { resolveRouteStructure } from "./Routes";
```

Comment:
Originally I envisioned this logic being part of the SideNav, and having this method execute whenever the input route list prop changed. However, because of how lit handles detecting prop mutation, I couldn't find a clear way of determining when to trigger this method. So for now, I've got it being executed independently of the element, and the results passed directly into the prop.

One option I was considering that would avoid change detection concerns was making a public method available on the element that could accept a parameter, and have that trigger route resolution. However I didn't see a supported way of doing that either.

Reply:
I figured out how to properly detect prop changes, so this is no longer needed. The `RouteTree` will be generated by the `SideNav` component when its `routes` prop is updated